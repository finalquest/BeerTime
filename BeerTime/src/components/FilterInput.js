import React, { useState } from 'react';


import {
  View,
  StyleSheet,
  TouchableOpacity,
} from 'react-native';

import PropTypes from 'prop-types';

import TextInput from './TextInput';
import Label from './Label';

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'white',
    alignSelf: 'stretch',
    marginHorizontal: 20,
    padding: 20,
  },
  input: {
    borderBottomWidth: 1,
    borderColor: 'lightgray',
    marginBottom: 5,
  },
  inputLabel: {
    fontSize: 12,
    marginTop: 5,
    marginBottom: 5,
    color: 'dimgrey',
  },
  title: {
    fontSize: 18,
    textAlign: 'left',
    fontWeight: '400',
    marginBottom: 20,
  },
  primary: {
    flex: 1,
    backgroundColor: '#4169ff',
    height: 45,
    alignSelf: 'stretch',
    marginTop: 10,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 5,
    marginHorizontal: 5,
  },
  text: {
    color: 'white',
    fontSize: 18,
  },
  secondary: {
    backgroundColor: 'white',
    borderColor: '#4169ff',
    borderWidth: 1,
  },
  textSecondary: {
    color: '#4169ff',
    fontSize: 18,
  },
});

const FilterInput = ({ onFilterSelected, onFilterCancel }) => {
  const [inputValue, setInputValue] = useState(false);

  return (
    <View style={styles.container}>
      <Label style={styles.title}>Search</Label>
      <TextInput
        label="Beer Name"
        style={styles.input}
        labelStyle={styles.inputLabel}
        value={inputValue}
        onChange={(text) => {
          setInputValue(text);
        }}
      />
      <View style={{ flexDirection: 'row', alignSelf: 'stretch' }}>
        <TouchableOpacity
          style={styles.primary}
          onPress={() => {
            onFilterSelected();
          }}
        >
          <Label style={styles.text}>OK</Label>
        </TouchableOpacity>
        <TouchableOpacity
          style={[styles.primary, styles.secondary]}
          onPress={() => {
            onFilterCancel();
          }}
        >
          <Label style={[styles.text, styles.textSecondary]}>Cancel</Label>
        </TouchableOpacity>
      </View>
    </View>
  );
};

FilterInput.propTypes = {
  onFilterSelected: PropTypes.func.isRequired,
  onFilterCancel: PropTypes.func.isRequired,
};

FilterInput.defaultProps = {
};

export default FilterInput;
