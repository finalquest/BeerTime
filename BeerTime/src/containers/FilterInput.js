import { connect } from 'react-redux';
import FilterInputScreen from '../screens/FilterInputScreen';


const mapStateToProps = () => ({
});
const mapDispatchToProps = () => ({
  onFilter: () => { console.log('filter'); },
});
export default connect(mapStateToProps, mapDispatchToProps)(FilterInputScreen);
