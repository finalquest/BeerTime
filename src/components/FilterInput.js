import React, { useState } from 'react';


import {
  View,
  StyleSheet,
  TouchableOpacity,
} from 'react-native';

import PropTypes from 'prop-types';

import TextInput from './TextInput';
import Label from './Label';

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'white',
    alignSelf: 'stretch',
    marginHorizontal: 20,
    padding: 20,
  },
  input: {
    borderBottomWidth: 1,
    borderColor: 'lightgray',
    marginBottom: 5,
  },
  inputLabel: {
    fontSize: 12,
    marginTop: 5,
    marginBottom: 5,
    color: 'dimgrey',
  },
  title: {
    fontSize: 18,
    textAlign: 'left',
    fontWeight: '400',
    marginBottom: 20,
  },
  primary: {
    flex: 1,
    backgroundColor: '#4169ff',
    height: 45,
    alignSelf: 'stretch',
    marginTop: 10,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 5,
    marginHorizontal: 5,
  },
  text: {
    color: 'white',
    fontSize: 18,
  },
  secondary: {
    backgroundColor: 'white',
    borderColor: '#4169ff',
    borderWidth: 1,
  },
  textSecondary: {
    color: '#4169ff',
    fontSize: 18,
  },
});

const FilterInput = ({ onFilterSelected, onFilterCancel }) => {
  const [inputValue, setInputValue] = useState(undefined);
  const [fromBrewDate, setFromBrewDate] = useState({ value: '' });
  const [toBrewDate, setToBrewDate] = useState({ value: '' });

  const validateInput = () => {
    const datePattern = /[\d]{2}-[\d]{4}$/;
    const { value: toBrewDateValue } = toBrewDate;
    const { value: fromBrewDateValue } = fromBrewDate;

    let error = false;

    if (toBrewDateValue !== '' && !datePattern.test(toBrewDateValue)) {
      setToBrewDate({ value: toBrewDateValue, error: 'Invalid format. Should be MM-YYYY' });
      error = true;
    }
    if (fromBrewDateValue !== '' && !datePattern.test(fromBrewDateValue)) {
      setFromBrewDate({ value: fromBrewDateValue, error: 'Invalid format. Should be MM-YYYY' });
      error = true;
    }

    if (!error) {
      onFilterSelected({
        name: inputValue,
        fromBrewDate: fromBrewDateValue,
        toBrewDate: toBrewDateValue,
      });
    }
  };
  return (
    <View style={styles.container}>
      <Label style={styles.title}>Search</Label>
      <TextInput
        label="Beer Name"
        style={styles.input}
        labelStyle={styles.inputLabel}
        value={inputValue}
        onChange={(text) => {
          setInputValue(text);
        }}
      />
      <TextInput
        label="After First Brew Date"
        style={styles.input}
        labelStyle={styles.inputLabel}
        value={fromBrewDate.value}
        error={fromBrewDate.error}
        onChange={(text) => {
          setFromBrewDate({ value: text });
        }}
      />
      <TextInput
        label="Before First Brew Date"
        style={styles.input}
        labelStyle={styles.inputLabel}
        value={toBrewDate.value}
        error={toBrewDate.error}
        onChange={(text) => {
          setToBrewDate({ value: text });
        }}
      />
      <View style={{ flexDirection: 'row', alignSelf: 'stretch' }}>
        <TouchableOpacity
          style={styles.primary}
          onPress={() => {
            validateInput();
            // onFilterSelected({ name: inputValue, fromBrewDate });
          }}
        >
          <Label style={styles.text}>OK</Label>
        </TouchableOpacity>
        <TouchableOpacity
          style={[styles.primary, styles.secondary]}
          onPress={() => {
            onFilterCancel();
          }}
        >
          <Label style={[styles.text, styles.textSecondary]}>Cancel</Label>
        </TouchableOpacity>
      </View>
    </View>
  );
};

FilterInput.propTypes = {
  onFilterSelected: PropTypes.func.isRequired,
  onFilterCancel: PropTypes.func.isRequired,
};

FilterInput.defaultProps = {
};

export default FilterInput;
