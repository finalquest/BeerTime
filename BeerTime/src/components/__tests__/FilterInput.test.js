import React from 'react';
import { shallow } from 'enzyme';

import FilterInput from '../FilterInput';

describe('FilterInput test suite', () => {
  const setState = jest.fn();
  const useStateSpy = jest.spyOn(React, 'useState');
  useStateSpy.mockImplementation(init => [init, setState]);

  it('renders as expected', () => {
    const wrapper = shallow(<FilterInput />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should set inputvalue state on textInput onChange', () => {
    const wrapper = shallow(<FilterInput />);
    wrapper.find('TextInput').prop('onChange')('test');
    expect(setState).toBeCalledWith('test');
  });

  it('should call callback if ok button pressed', () => {
    const selected = jest.fn();
    const wrapper = shallow(<FilterInput onFilterSelected={selected} />);
    wrapper.find('TouchableOpacity').at(0).prop('onPress')();
    expect(selected).toBeCalled();
  });

  it('should call callback if ok button pressed', () => {
    const selected = jest.fn();
    const wrapper = shallow(<FilterInput onFilterCancel={selected} />);
    wrapper.find('TouchableOpacity').at(1).prop('onPress')();
    expect(selected).toBeCalled();
  });
});
