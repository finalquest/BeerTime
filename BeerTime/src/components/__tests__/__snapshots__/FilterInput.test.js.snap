// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterInput test suite renders as expected 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "white",
      "marginHorizontal": 20,
      "padding": 20,
    }
  }
>
  <Label
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 18,
        "fontWeight": "400",
        "marginBottom": 20,
        "textAlign": "left",
      }
    }
    uppercase={false}
  >
    Search
  </Label>
  <TextInput
    keyboardType="default"
    label="Beer Name"
    labelStyle={
      Object {
        "color": "dimgrey",
        "fontSize": 12,
        "marginBottom": 5,
        "marginTop": 5,
      }
    }
    onChange={[Function]}
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "lightgray",
        "marginBottom": 5,
      }
    }
  />
  <TextInput
    keyboardType="default"
    label="After First Brew Date"
    labelStyle={
      Object {
        "color": "dimgrey",
        "fontSize": 12,
        "marginBottom": 5,
        "marginTop": 5,
      }
    }
    onChange={[Function]}
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "lightgray",
        "marginBottom": 5,
      }
    }
    value=""
  />
  <TextInput
    keyboardType="default"
    label="Before First Brew Date"
    labelStyle={
      Object {
        "color": "dimgrey",
        "fontSize": 12,
        "marginBottom": 5,
        "marginTop": 5,
      }
    }
    onChange={[Function]}
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "lightgray",
        "marginBottom": 5,
      }
    }
    value=""
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "flexDirection": "row",
      }
    }
  >
    <TouchableOpacity
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#4169ff",
          "borderRadius": 5,
          "flex": 1,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "marginHorizontal": 5,
          "marginTop": 10,
        }
      }
    >
      <Label
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 18,
          }
        }
        uppercase={false}
      >
        OK
      </Label>
    </TouchableOpacity>
    <TouchableOpacity
      onPress={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "#4169ff",
            "borderRadius": 5,
            "flex": 1,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "center",
            "marginHorizontal": 5,
            "marginTop": 10,
          },
          Object {
            "backgroundColor": "white",
            "borderColor": "#4169ff",
            "borderWidth": 1,
          },
        ]
      }
    >
      <Label
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 18,
            },
            Object {
              "color": "#4169ff",
              "fontSize": 18,
            },
          ]
        }
        uppercase={false}
      >
        Cancel
      </Label>
    </TouchableOpacity>
  </View>
</View>
`;
